wger/nutrition/dataclasses.py

```diff
@@ -64,8 +68,44 @@ def sanity_checks(self):
             if value and value > 100:
                 raise ValueError(f'Value for {macro} is greater than 100: {value}')
 
+        if self.fat_saturated and self.fat_saturated > self.fat:
+            raise ValueError(
+                f'Saturated fat is greater than fat: {self.fat_saturated} > {self.fat}'
+            )
+
+        if self.carbohydrates_sugar and self.carbohydrates_sugar > self.carbohydrates:
+            raise ValueError(
+                f'Sugar is greater than carbohydrates: {self.carbohydrates_sugar} > {self.carbohydrates}'
+            )
+
         if self.carbohydrates + self.protein + self.fat > 100:
             raise ValueError(f'Total of carbohydrates, protein and fat is greater than 100!')
 
+        # Energy approximations
+        energy_protein = self.protein * ENERGY_FACTOR['protein']['metric']
+        energy_carbohydrates = self.carbohydrates * ENERGY_FACTOR['carbohydrates']['metric']
+        energy_fat = self.fat * ENERGY_FACTOR['fat']['metric']
+        energy_calculated = energy_protein + energy_carbohydrates + energy_fat
+
+        if energy_fat > self.energy:
+            raise ValueError(
+                f'Energy calculated from fat is greater than total energy: {energy_fat} > {self.energy}'
+            )
+
+        if energy_carbohydrates > self.energy:
+            raise ValueError(
+                f'Energy calculated from carbohydrates is greater than total energy: {energy_carbohydrates} > {self.energy}'
+            )
+
+        if energy_protein > self.energy:
+            raise ValueError(
+                f'Energy calculated from protein is greater than total energy: {energy_protein} > {self.energy}'
+            )
```

Comment:
The protein, carb and fat energy values are rough. Checking this too strictly might trigger false positives , especially for ingredients that only contain 1 macro (eg egg whites)

For sanity checking maybe allow 20% error or so ? I would be curious to have a look at ingredients for which we trigger errors here

Reply:
Also all these checks are quite verbose. Do we need to raise ValueErrors explicitly? Could we replace these with simple, single line assert statements ? If the code is more obvious, it also reduces the need for unit testing